integration/integration_test.go

```diff
@@ -313,3 +313,66 @@ func TestDemoteLeader(t *testing.T) {
 	numManager--
 	pollClusterReady(t, cl, numWorker, numManager)
 }
+
+func TestJoinDemoteRace(t *testing.T) {
```

Comment:
This seems to increase the runtime of the integration tests from about 70 seconds to about 120 seconds :(

Wondering if we can do anything to speed it up.

Are the three iterations in the loop below worthwhile?

Is there anything we can do to speed up the actual demotion process, or whatever else is taking a long time in the test?